### PR TITLE
Fix atmo resource request

### PR DESCRIPTION
**Pull request by cdosborn**
This is related to [ATMO-1415](https://pods.iplantcollaborative.org/jira/browse/ATMO-1415). This re-enables approve/deny resource requests. Mainly the changes are relevant to the denying action which hadn't been working for a while.

This is ready to be reviewed, but it should not be merged until this [tropo PR](https://github.com/iPlantCollaborativeOpenSource/troposphere/pull/408) is ready.

I'm concerned that these changes may affect `MachineRequest`. 

Easily checkout this pr with:
```
hub checkout https://github.com/iPlantCollaborativeOpenSource/atmosphere/pull/167
```

### CODE REVIEW

File: core/email.py [hunk 1]
```diff
@@ -310,9 +311,13 @@ def send_denied_resource_email(user, request, reason):
...
-    body = render_to_string("core/email/resource_request_denied.html",
-                            context=Context(context))
-    return email_from_admin(user, subject, body)
+    from_name, from_email = admin_address()
+    user_email = lookupEmail(user.username)
+    recipients = [email_address_str(user.username, user_email)]
+    sender = email_address_str(from_name, from_email)
+
+    return send_email_template(subject, template, recipients, sender,
+                               context=context, cc=[sender])
```
Comment (cdosborn) [inline core/email.py]: This code is copied from [above](https://github.com/iPlantCollaborativeOpenSource/atmosphere/pull/167/commits/5b3a415328a42bf4a2a87783e4943133e3f2fe34#diff-8d164c7b0e814633663ff40f14c113aeL275)

File: core/migrations/0055_update_templates_with_cyverse.py [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+
+def try_update_iplant_to_cyverse(apps, schema_editor):
+    try:
+        EmailTemplate = apps.get_model("core", "EmailTemplate")
+        template = EmailTemplate.objects.get(email_address='support@iplantcollaborative.org')
+        template.email_address = u'support@cyverse.org'
+        template.email_footer = u'CyVerse Atmosphere Team'
+        template.save()
+    except:
+        pass
```
Comment (cdosborn) [inline core/migrations/0055_update_templates_with_cyverse.py]: I tested this with bogus values that would throw the exception. This is safe to run on jetstream.

**Comment by cdosborn** [comment]
This is ready to be reviewed.

File: api/v2/views/resource_request.py [hunk 3]
```diff
@@ -66,9 +66,9 @@ def deny_action(self, instance):
...
         """
-        self.end_date = timezone.now()
-        self.save()
+        instance.end_date = timezone.now()
+        instance.save()
         email.send_denied_resource_email(
             user=instance.created_by,
             request=instance.request,
-            reason=instance.admin_message)
+            reason=instance.admin_message).delay()
```
Comment (steve-gregory) [inline api/v2/views/resource_request.py]: apply_async?

File: core/models/quota.py [hunk 4]
```diff
@@ -146,12 +146,6 @@ def default_dict(cls):
     class Meta:
         db_table = 'quota'
         app_label = 'core'
-        # unique_together = ("cpu", "memory", "storage",
-        #                    "floating_ip_count",
-        #                    "port_count",
-        #                    "instance_count",
-        #                    "snapshot_count",
-        #                    "storage_count")
```
Comment (steve-gregory) [inline core/models/quota.py]: No more uniqueness tuples?

File: api/v2/views/resource_request.py (hunk 3 above)
Comment (cdosborn) [inline api/v2/views/resource_request.py]: So prior to this `.delay` the task was not being performed. I should change the task to not be async?

File: core/models/quota.py (hunk 4 above)
Comment (cdosborn) [inline core/models/quota.py]: NO MORE! (this isn't the commit that disable them btw, i'm just deleting the comments).

I removed them for allocation as well. There's no reason to force them to be unique. It adds the overhead of having to check if they exist before assigning them to a user. This is going to make for intense queries later on.

**Comment by steve-gregory** [comment]
Updating migrations to drop this PR in after #182, #180 ... This PR will be :shipit: shortly..

**Pull request merged**